helix-view/src/document.rs

```diff
@@ -397,17 +398,103 @@ impl Document {
 
     /// The same as [`format`], but only returns formatting changes if auto-formatting
     /// is configured.
-    pub fn auto_format(&self) -> Option<impl Future<Output = LspFormatting> + 'static> {
+    pub fn auto_format(
+        &mut self,
+        view_id: ViewId,
+    ) -> Option<impl Future<Output = LspFormatting> + 'static> {
         if self.language_config()?.auto_format {
-            self.format()
+            self.format(view_id)
         } else {
             None
         }
     }
 
     /// If supported, returns the changes that should be applied to this document in order
     /// to format it nicely.
-    pub fn format(&self) -> Option<impl Future<Output = LspFormatting> + 'static> {
+    pub fn format(
+        &mut self,
+        view_id: ViewId,
+    ) -> Option<impl Future<Output = LspFormatting> + 'static> {
```

Comment:
Changing the return type from `Option` to `Result` would be more appropriate here now, since multiple things could go wrong here and the there would be no feedback on the screen it anything errors out. Any error can be displayed in the editor by the calling functions instead of being simply logged.

Reply:
Would I just remove the `.ok()` and return the errors directly like that? Or should I make errors specific like when using `log::error!` ?

Reply:
We definitely need the context added by the log commands, so you could use `anyhow::Result` for the return type and `result.map_err(|e| anyhow!("previous log description"))?` where ever `option.ok()` was being used.